### PR TITLE
front: add placeholder 0 in budget and allowances inputs

### LINKED ISSUE #5221: Numeric field should have place holders and not default value
For instance in these fields:

Image

The `0` is a default value not a placeholder, so it is immpossible to remove the `0` and if you want a `3%` margin you will have either `03` or `30` and then remove the 0.

List of the concerned inputs
- [x] #5548 
- [x] #5549
- [x] #5550 

Version: cb78a63b

### LINKED ISSUE #5477: Number format in budget field
### What happened?

Image

### What did you expect to happen?

When you type a amount, you don't expect a 0 in front of the number

### How can we reproduce it (as minimally and precisely as possible)?

Create a project and fil the fields budget
Delete the budget and type a new budget if necessary, a zero remain in front 

### What operating system, browser and environment are you using?

- **Browser**: Version 118.0.5993.96 
- **OS**: Sonoma 14.0 (23A344)
- **Env**: rec-osrd.reseau.sncf.fr
- 

### OSRD version (top right corner `Account` button > `Informations`)

1280353d6bb501ba1b4ad3c1ffe91927c6a6cfd0

### LINKED ISSUE #5548: front: add placeholders to budget and allowances inputs

### PR DESCRIPTION
**Pull request by clarani**
closes #5548 
closes #5477

- created new types AllowanceValueForm in which value can be undefined
- added placeholders in allowances inputs, as well as in budget inputs

image
image

### CODE REVIEW

File: front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/types.ts [hunk 1]
```diff
@@ -1,16 +1,49 @@
-import { EngineeringAllowance, RangeAllowance } from 'common/api/osrdEditoastApi';
-
 export enum AllowancesTypes {
   standard = 'standard',
   engineering = 'engineering',
 }
+
+export type AllowanceValueForm =
```
Comment (anisometropie) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/types.ts:6]: this already exists : src/common/api/osrdEditoastApi.ts:2057

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/helpers.ts [hunk 2]
```diff
@@ -31,15 +34,15 @@ export function findAllowanceOverlap({
     : [-1, -1];
 }
 
-export default function getAllowanceValue(value?: AllowanceValue) {
-  switch (value?.value_type) {
+export default function getAllowanceValue(value: AllowanceValueForm) {
```
Comment (anisometropie) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/helpers.ts:37]: why do you want to introduce potentially undefined values and the need to deal with them everytime it,s called ?
the 0 returned here **is** the placeholder

File: front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/AllowancesActions.tsx [hunk 3]
```diff
@@ -74,10 +72,15 @@ export default function AllowancesActions({
...
   const [allowanceLength, setAllowanceLength] = useState(endPosition - beginPosition);
   const [distribution, setDistribution] = useState(distributionsList[0].value);
-  const [valueAndUnit, setValueAndUnit] = useState<AllowanceValue>();
+  const [valueAndUnit, setValueAndUnit] = useState<AllowanceValueForm>();
   const [isValid, setIsValid] = useState(false);
 
-  function validityTest() {
+  const allowanceValue = useMemo(() => {
+    if (valueAndUnit) return getAllowanceValue(valueAndUnit);
+    return undefined;
```
Comment (anisometropie) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/AllowancesActions.tsx:80]: the function is considered to return undefined if there is no return statement

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/AllowancesActions.tsx (hunk 3 above)
Comment (clarani) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/AllowancesActions.tsx:80]: I have this error if I remove the return undefined, so I will keep it like this if it's ok for you
image

Comment (anisometropie) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/AllowancesActions.tsx:80]: yes

**Review by SharglutDev — COMMENTED** [review]
Getting this in all inputs where you are using undefined as default value

image

**Review by SharglutDev — APPROVED** [review]

**Review by anisometropie — APPROVED** [review]

File: front/public/locales/en/translation.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
     "back": "Back",
     "cancel": "Cancel",
     "change": "Change",
+    "choose": "choose…",
```
Comment (Akctarus) [inline front/public/locales/en/translation.json]:
```suggestion
    "choose": "Choose…",
```

File: front/public/locales/fr/translation.json [hunk 5]
```diff
@@ -5,6 +5,7 @@
     "back": "Retour",
     "cancel": "Annuler",
     "change": "Changer",
+    "choose": "choisissez…",
```
Comment (Akctarus) [inline front/public/locales/fr/translation.json]:
```suggestion
    "choose": "Choisissez…",
```

File: front/public/locales/en/translation.json
Comment (Akctarus) [inline front/public/locales/en/translation.json:1]: for uniformity, I think it should be capitalized.

File: front/public/locales/en/translation.json (hunk 4 above)
Comment (Akctarus) [reply front/public/locales/en/translation.json]: for uniformity, I think it should be capitalized.

File: front/public/locales/fr/translation.json (hunk 5 above)
Comment (Akctarus) [reply front/public/locales/fr/translation.json]: same here

File: front/public/locales/en/translation.json (hunk 4 above)
Comment (clarani) [reply front/public/locales/en/translation.json]: Thibaut asked specifically for lowercase, so I think we should keep it like this (but I agree that I was surprised at first when he asked for lowercase :+1: )

File: front/public/locales/fr/translation.json (hunk 5 above)
Comment (clarani) [reply front/public/locales/fr/translation.json]: See comment above

File: front/public/locales/en/translation.json (hunk 4 above)
Comment (Akctarus) [reply front/public/locales/en/translation.json]: Ok perfect, I didn't know that !

**Review by Akctarus — APPROVED** [review]
LGTM

**Comment by clarani** [comment]
Viewed with Nicolas and Thibaut, we decided to simply remove  the placeholders (instead of placeholders "choisissez..."

**Review by nicolaswurtz — APPROVED** [review]

**Pull request merged**